cavern/src/main/java/org/opencadc/cavern/nodes/CephFSQuotaPlugin.java

```diff
@@ -0,0 +1,158 @@
+    public Long getBytesUsed(Path directory) {
+        LOGGER.debug("getBytesUsed: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot get size of null directory");
+        }
+
+        try {
+            final String recursiveBytesUsedValue =
+                    ExtendedFileAttributes.getFileAttribute(directory, CephFSQuotaPlugin.RECURSIVE_SIZE_ATTR_KEY,
+                                                            CephFSQuotaPlugin.NAMESPACE);
+            final long recursiveBytesUsed = Long.parseLong(recursiveBytesUsedValue);
+            LOGGER.debug("getBytesUsed: " + directory + ": OK");
+            return recursiveBytesUsed;
+        } catch (IOException ioException) {
+            LOGGER.warn("Unable to find recursive folder size of " + directory + " ("
+                        + CephFSQuotaPlugin.NAMESPACE + "." + CephFSQuotaPlugin.QUOTA_ATTR_KEY + ")");
+            return null;
+        }
+    }
+
+    /**
+     * Obtain the quota amount, in bytes, set at the highest level of the provided path.  This method will traverse,
+     * beginning at the first Path element, until it finds a directory containing the CephFS Quota attribute.
+     * Returning a null value means no quota is set, and users can write until the underlying storage is full.
+     *
+     * @param directory directory to check
+     * @return null if the directory input is null, or if no directory contains the quota attribute.
+     */
+    @Override
+    public Long getQuota(Path directory) {
+        LOGGER.debug("getQuota: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot check quota of null directory");
+        }
+
+        try {
+            final String quotaAttributeValue = ExtendedFileAttributes.getFileAttribute(directory,
+                                                                                       CephFSQuotaPlugin.QUOTA_ATTR_KEY,
+                                                                                       CephFSQuotaPlugin.NAMESPACE);
+            final Long quotaInBytes = StringUtil.hasText(quotaAttributeValue)
+                                      ? Long.parseLong(quotaAttributeValue) : null;
+            LOGGER.debug("getQuota: " + directory + ": OK");
+
+            return quotaInBytes;
+        } catch (IOException ioException) {
+            LOGGER.warn("Unable to find a folder containing " + CephFSQuotaPlugin.NAMESPACE + "."
+                        + CephFSQuotaPlugin.QUOTA_ATTR_KEY + " from " + directory);
+            return null;
+        }
+    }
+
+    @Override
+    public void setQuota(Path directory, Long quota) {
+        LOGGER.debug("setQuota: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Attempt to set quota on null directory.  Doing nothing.");
+        }
+
+        final String quotaValue = (quota == null || quota == 0L) ? null : Long.toString(quota);
```

Comment:
w.r.t. quota == 0: I do recall readiung something about using 0 for the ceph attr, but that should be an implementation detail. The only way to remove the quota should be calling this methid with `quota == null`. That's the method API here. 


I don't know if we have a use case to set  a quota to 0 (completely block writes) so I might even throw an IllegalArgumentException if `quota <= 0` for now, just to make it clearly invalid.